fem/hyperbolic.hpp

```diff
@@ -414,20 +418,18 @@ class HyperbolicFormIntegrator : public NonlinearFormIntegrator
 /**
  * @brief Rusanov flux, also known as local Lax-Friedrichs,
  *    F̂ n = ½(F(u⁺,x)n + F(u⁻,x)n) - ½λ(u⁺ - u⁻)
- * where λ is the maximum characteristic velocity
- * @note The implementation assumes monotonous F(u,x) in u
+ * where λ = max(|F(u⁺,x)n|, |F(u⁻,x)n|).
+ * @note This construction assumes monotonous F(u,x) in u,
+ * implying λ is the maximum characteristic velocity.
```

Comment:
This does not make sense for systems.  I've given suggestions how to remedy this.

Reply:
@dohyun-cse should correct me if I'm wrong, but what the current implementation of Rusanov / (local) Lax Friedrichs flux does assume is that the maximum characteristic velocity is given by $max( |\lambda(u_-)|, |\lambda(u_+)|)$ or in other words, that the maximum characteristic velocity overall (or between $u_-$ and $u_+$) is found by evaluating the maximum characteristic velocity at $u_-$ and $u_+$.  Note the maximum characteristic velocity at $u_\pm$ is **not** $F(u_\pm)$** as introduced in 5895c08.  This needs to be remedied.

I suggest simply stating that the $\max |\frac{df}{du}(u)|$ quantity in the Rusanov / (local) Lax Friedrichs flux is assumed to be $\max_{u = u_\pm} |\frac{df}{du}(u)|$

Reply:
Ahh, yes, we are talking about the (characteristic) velocity, so it is the derivative. I just copied that over from elsewhere. I need to fix that everywhere.
Regarding systems, it makes some sense, because it should be the maximal velocity (but what it really is depends on the `FluxFunction`) and a scalar function can be monotonous in multiple parameters.

Reply:
Marking thread as resolved as the line referenced is now outdated.